### PR TITLE
Add support for sticker pack extension & app extension products

### PR DESCRIPTION
**Pull request by Rag0n**
There is no issue, but I'm going to integrate tuist into my workflow. The lack of ability to generate sticker pack extensions and app extensions stand on my way.

### Short description 📝

Adds ability to generate sticker pack extensions and app extensions.

### Solution 📦

I tried to be consistent with code base, so solution looks similar to other code related to product types and build phase generation.

If solution looks good I will write unit tests. Should I write fixture aswell?

### CODE REVIEW

File: Sources/ProjectDescription/Product.swift [hunk 1]
```diff
@@ -21,5 +21,5 @@ public enum Product: String, Codable {
 //    case tvExtension
 //    case messagesApplication
 //    case messagesExtension
-//    case stickerPack
+    case stickerPack = "sticker_pack"
```
Comment (kwridan) [inline Sources/ProjectDescription/Product.swift]: perhaps here `stickersPackExtension` ? (to help distinguish it from `stickersPackApp`)

Comment (pepibumur) [reply Sources/ProjectDescription/Product.swift]: Agree here

File: Sources/TuistGenerator/Graph/Graph.swift [hunk 2]
```diff
@@ -321,6 +322,19 @@ class Graph: Graphing {
         return Set(references).sorted()
     }
 
+    func appExtensionDependencies(path: AbsolutePath, name: String) -> [TargetNode] {
```
Comment (pepibumur) [inline Sources/TuistGenerator/Graph/Graph.swift]: Beautifully implemented. What about adding a test to this method?

File: Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift [hunk 3]
```diff
@@ -253,4 +260,27 @@ final class BuildPhaseGenerator: BuildPhaseGenerating {
             resourcesBuildPhase.files?.append(pbxBuildFile)
         }
     }
+
+    func generateAppExtensionsBuildPhase(path: AbsolutePath,
```
Comment (pepibumur) [inline Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift]: What about adding an unit test here?

File: Sources/ProjectDescription/Product.swift (hunk 1 above)
Comment (Rag0n) [reply Sources/ProjectDescription/Product.swift]: Sure, by the way there is no stickersPackApp currently in XcodeProj

Comment (pepibumur) [reply Sources/ProjectDescription/Product.swift]: 😬 we'll have to add it at some point. Feel free to open a PR there :)

**Comment by pepibumur** [comment]
Hey @Rag0n, just following up on this PR. Let us know if you would like any of us to do a final push to this PR and get it merged.  We'd be happy to do so.

**Comment by Rag0n** [comment]
Hi, I'm sorry for my silence and absence, I will update PR today

**Comment by Rag0n** [comment]
> Based on your experience with sticker pack targets are there any rules/gotchas worth linting for? (e.g. must only contain resources / must have certain settings? etc...)

StickerPackExtension should not contain any source files and I have added that rule. I don't know if any other rules exists.

**Comment by Rag0n** [comment]
I added support for app extensions and wrote tests. I think fixtures are the only thing is left.

**Comment by kwridan** [comment]
Nice work @Rag0n ! 👍 

This PR nearly ready, just need a run of `swiftformat .` to fix formatting issues, and looks like one of the tests needs a small update (`ProductTests.test_forPlatform_when_ios`).

do you need a hand with the fixtures?

**Comment by kwridan** [comment]
Hi @Rag0n - I would like add more extensions support and this PR would make a great base for those changes. Happy to help with finishing this PR.

**Comment by Rag0n** [comment]
Hi @kwridan, sure, I’m too slow anyway :)

**Pull request merged**